src/legacy/core_plugins/kibana/public/management/sections/settings/components/advanced_settings_voice_announcement/advanced_settings_voice_announcement.tsx

```diff
@@ -40,9 +40,15 @@
 import React, { Component } from 'react';
 import { FormattedMessage } from '@kbn/i18n/react';
 import { EuiScreenReaderOnly, EuiDelayRender } from '@elastic/eui';
+import { FieldSetting } from '../../types';
 
-export class AdvancedSettingsVoiceAnnouncement extends Component {
-  shouldComponentUpdate = nextProps => {
+interface Props {
+  queryText: string;
+  settings: Record<string, FieldSetting[]>;
+}
+
+export class AdvancedSettingsVoiceAnnouncement extends Component<Props> {
```

Comment:
I think this class could be really trivially turned into a functional component.

Reply:
I'm happy to do that, but how should I account for the `shouldComponentUpdate` functionality?